### PR TITLE
PP-231: qstat performance improvement 

### PR DESCRIPTION
**Pull request by vinodchitrali**
#### Issue-ID

* PP-231

#### Problem description
* Qstat is taking too much time when querying for a set of jobs

#### Cause / Analysis
* qstat command was connecting to PBS server for every job id

#### Solution description
* Passing all job ids to server at once when used with -E option

#### Checklist:

- [x] I have joined the **[pbspro community forum](http://community.pbspro.org/)**.
- [x] My pull request contains a **single, signed** commit. See **[setting up gpg signature](https://pbspro.atlassian.net/wiki/display/DG/Signing+Your+Git+Commits)**.
- [x] My code follows the **[coding style](https://pbspro.atlassian.net/wiki/display/DG/Coding+Standards)** of this project.
- [x] My change requires project documentation. See **[required documentation checklist](https://pbspro.atlassian.net/wiki/display/DG/Checklist+for+Developing+Features+and+Bug+Fixes)** for details.
- [x] I have added documentation in the **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)**.
- [x] I have added tests to cover my changes. To create automated tests, see **[using PTL for testing](https://pbspro.atlassian.net/wiki/display/DG/Using+PTL+for+Testing)**.
- [x] All new and existing automated tests have passed. See **[running automated PTL tests](https://pbspro.atlassian.net/wiki/display/DG/PTL+Quick+Start+Guide)**.
- [x] I have attached automated (PTL)/manual test logs to the associated Jira ticket.

__***For further information please visit the [Developer Guide Home](https://pbspro.atlassian.net/wiki/display/DG/Developer+Guide+Home).***__

### CODE REVIEW

File: src/cmds/qstat.c [hunk 1]
```diff
@@ -2223,7 +2322,56 @@ qstat -B [-f] [ server_name... ]\n";
...
+								free(query_job_list);
+								query_job_list = strdup(job_list);
+								memset(job_list, 0, ((argc-1) * (PBS_MAXCLTJOBID+1)));
+								strncpy(job_list, job_id_out, PBS_MAXCLTJOBID-1);
+								strncat(job_list, ",", 1);
+							}
+						}
+					}
 				} else {  /* must be a destination-id */
+					if (E_opt == 1) {
```
Comment (dilip-krishnan) [inline src/cmds/qstat.c]: Hi Vinod, 
     This behaviour is not documented in the EDD yet. Changes Look fine to me
I sign-off, if this behaviour is going get written in the EDD.

**Comment by suresh-thelkar** [comment]
Code changes look good to me.

File: src/cmds/qstat.c (hunk 1 above)
Comment (vinodchitrali) [inline src/cmds/qstat.c]: Updated EDD
https://pbspro.atlassian.net/wiki/display/PD/PP-231%3A++qstat+express+option

**Comment by subhasisb** [comment]
All green, merging.

**Pull request merged**